python/pyspark/tests/test_context.py

```diff
@@ -246,6 +247,20 @@ def test_startTime(self):
         with SparkContext() as sc:
             self.assertGreater(sc.startTime, 0)
 
+    def test_forbid_insecure_gateway(self):
+        # Fail immediately if you try to create a SparkContext
+        # with an insecure gateway
+        gateway = _launch_gateway(insecure=True)
```

Comment:
This test seems to be just for the check you're adding in the `SparkContext` python class, right?

So instead of this `insecure` flag, couldn't you just instantiate a dummy insecure gateway? Then the Java and Python code could be a little simpler and avoid the test-only checks.

Reply:
oh great point, that simplified this code a ton.